wikiclass/feature_lists/wikidata/datasources/revision_oriented.py

```diff
@@ -0,0 +1,267 @@
+			list_of_claims = claim
+			for i, source in enumerate(list_of_claims.sources):
+				for index_list_1, index_list_2 in source.items(): 
+					sources_in_JSON = index_list_2[0].toJSON()
+					list_sources_in_JSON.append("property : " + str(sources_in_JSON['property']) + " & value : " + str(sources_in_JSON['datavalue']['value']))
+	
+	#eliminate duplicates
+	for value in list_sources_in_JSON:
+		result_set.add(value)
+	
+	return result_set
+	
+
+def _process_complete_translations(item):
+	item_labels_dict = item.labels
+	item_desc_dict = item.descriptions
+	combined_dict = {}
+	result_set = []
+	
+	#merging item label dictionary and item description dictionary
+	for key in (item_labels_dict.keys() | item_desc_dict.keys()):
+		if key in item_labels_dict: combined_dict.setdefault(key, []).append(item_labels_dict[key])
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if a language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if(len(value[1]) == 2): 
+			result_set.append(value[0])
+			
+	return result_set
+
+def _process_important_translations(item):
+	item_labels_dict = item.labels
+	item_desc_dict = item.descriptions
+	combined_dict = {}
+	result_set = []
+	
+	#merging item label dictionary and item description dictionary
+	for key in (item_labels_dict.keys() | item_desc_dict.keys()):
+		if key in item_labels_dict: combined_dict.setdefault(key, []).append(item_labels_dict[key])
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if an important language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if((len(value[1]) == 2) and (value[0] == 'en' or value[0] == 'de' or value[0] == 'ar' or value[0] == 'zh' or value[0] == 'es' or value[0] == 'pt' or value[0] == 'ru' or value[0] == 'fr')): 
+			result_set.append(value[0])
+
+	return len(result_set)/8
+
+def _process_image_quality(item):
+
+    image_filename = ''
+    claims = list(set(
+        (property, _claim_to_str(claim)) 
+        for property in item.claims
+        for claim in item.claims[property]) 
+    )
+
+    try:
+        #find the image filename in P18 (image)
+        for x in claims:
+            if(x[0] ==  'P18'):
+                image_filename = x[1]
+                break
+
+        image_filename = image_filename.replace(' ', '_').strip()
+
+        #access the Commons API, retrieve the JSON response		
+        params = {'action': 'query', 'titles': 'Image:'+ image_filename, 'prop': 'imageinfo', 'iiprop': 'dimensions', 'iimetadataversion': 'latest','format': 'json'}
+        url = "https://commons.wikimedia.org/w/api.php?{}".format(urlencode(params))
+        url_request = urllib.request.urlopen(url)
+        data = url_request.read()
+        json_result = json.loads(data.decode("utf-8"))
```

Comment:
This isn't the right way to include a request to commons.  You want to incorporate this into `revscoring.extractors.api`

See https://www.wikidata.org/w/api.php?action=query&prop=pageimages&titles=Q42&piprop=original